### DISCUSSION TITLE
Why are there distinct autostart maps in PluginConfigurationSection and BundleLauncherHelper?

### DISCUSSION THREAD

**Discussion by haubi** [eclipse-pde/eclipse.pde#754] [General]
So I've stumbled over two Maps with different content that IMHO serve the same purpose - which seems to be:
"hold well-known system bundles that require non-default OSGi start parameters".
Per current master as of dfb96f0ce0a063ad3ea098e55587d4d4b97db3f9, these Maps are:
-  [org.eclipse.pde.internal.ui.editor.product.PluginConfigurationSection.RECOMMENDED_AUTOSTART_BUNDLES](https://github.com/eclipse-pde/eclipse.pde/blob/dfb96f0ce0a063ad3ea098e55587d4d4b97db3f9/ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/product/PluginConfigurationSection.java#L134C44-L134C73)
- [org.eclipse.pde.internal.launching.launcher.BundleLauncherHelper.AUTO_STARTED_BUNDLE_LEVELS](https://github.com/eclipse-pde/eclipse.pde/blob/dfb96f0ce0a063ad3ea098e55587d4d4b97db3f9/ui/org.eclipse.pde.launching/src/org/eclipse/pde/internal/launching/launcher/BundleLauncherHelper.java#L541C43-L541C69)

While I'm not necessarily wondering why there are two of them (might be a matter of "get around to ditch one"), I'm wondering whether there's a good reason for them to have different content, given that both have been touched somewhat recently.

What do I miss here?

Thanks!

**Comment by HannesWell** [comment]
> While I'm not necessarily wondering why there are two of them (might be a matter of "get around to ditch one"), I'm wondering whether there's a good reason for them to have different content, given that both have been touched somewhat recently.

I assume that those two locations initially had the same content (I can also only guess why they were split) and their content diverged over time.
With https://github.com/eclipse-pde/eclipse.pde/pull/644 there is already a PR to unify those.
It also adds the Apache Aries SPI-Fly bundle to the list of recommended bundles, but I'm currently a bit reluctant to add them since I hope to land https://github.com/eclipse-equinox/equinox/pull/295 soon, which provides an own OSGi Service Loader Mediator implementation in Equinox. Aries SPI-Fly would then be obsolete and I don't want to have that many changes in the list of recommendations.

But I will probably move the unification into a separate PR that can be submitted earlier.

**Comment by haubi** [comment]
Ok, everything answered, thanks!
